examples/retrieve-token/src/http/client.rs

```diff
@@ -0,0 +1,107 @@
+//! # Helpers to build a reqwest blocking client and handle responses and handle responses
+//!
+use http::Response as HttpResponse;
+use http::{Request, Response};
+use nr_auth::http_client::HttpClient as OauthHttpClient;
+use nr_auth::http_client::HttpClientError as OauthHttpClientError;
+use reqwest::blocking::{Client, Response as BlockingResponse};
+use reqwest::tls::TlsInfo;
+use std::time::Duration;
+
+const DEFAULT_AUTHENTICATOR_TIMEOUT: Duration = Duration::from_secs(5);
+
+#[derive(Debug, Clone)]
+pub struct HttpClient {
```

Comment:
not for this case, but i was jsut thinking if could be possible to define a trait implementation as a separate crate inside examples that could be reused in other examples 🤔 I guess that in some repo could make sense